src/sentry/sentry_metrics/querying/api.py

```diff
@@ -0,0 +1,314 @@
+            group_metrics = intermediate_groups.setdefault(group_key, {})
+            metric_values = group_metrics.setdefault(query_result.name, [[], 0])
+            # The item at position 1 is the "totals".
+            metric_values[1] = totals_item.get("aggregate_value")
+
+        meta = query_result.result["meta"]
+        for meta_item in meta:
+            meta_name = meta_item.get("name")
+            meta_type = meta_item.get("type")
+
+            # Since we have to handle multiple time series, we map the aggregate value to the actual
+            # metric name that was queried.
+            if meta_name == "aggregate_value":
+                intermediate_meta[query_result.name] = meta_type
+            else:
+                intermediate_meta[meta_name] = meta_type
+
+    translated_groups = []
+    for group_key, group_metrics in sorted(intermediate_groups.items(), key=lambda v: v[0]):
+        # This case should never happen, since if we have start and intervals not None
+        assert start is not None and end is not None and intervals is not None
+
+        start_seconds = int(start.timestamp())
+        num_intervals = len(intervals)
+
+        series = {}
+        totals = {}
+        for metric_name, metric_values in sorted(group_metrics.items(), key=lambda v: v[0]):
+            series[metric_name] = _generate_full_series(
+                start_seconds, num_intervals, interval, metric_values[0]
+            )
+            totals[metric_name] = metric_values[1]
+
+        inner_group = {
+            "by": {name: value for name, value in group_key},
+            "series": series,
+            "totals": totals,
+        }
+
+        translated_groups.append(inner_group)
+
+    translated_meta = [
+        {"name": meta_name, "type": meta_type} for meta_name, meta_type in intermediate_meta.items()
+    ]
+
+    return {
+        "intervals": intervals,
+        "groups": translated_groups,
+        "meta": translated_meta,
+        "start": start,
+        "end": end,
+    }
+
+
+def _build_request(
+    organization: Organization, use_case_id: UseCaseID, query: MetricsQuery
+) -> Request:
+    # TODO(layer): The layer should handle the dataset selection automatically.
+    dataset = Dataset.Metrics if use_case_id == UseCaseID.SESSIONS else Dataset.PerformanceMetrics
+    return Request(
+        dataset=dataset.value,
+        query=query,
+        app_id="default",
+        tenant_ids={"referrer": "metrics.data", "organization_id": organization.id},
+    )
+
+
+def _execute_series_and_totals_query(
+    organization: Organization, use_case_id: UseCaseID, interval: int, base_query: MetricsQuery
+) -> Mapping[str, Any]:
+    query = base_query.set_rollup(Rollup(interval=interval, granularity=_get_granularity(interval)))
+    series_result = run_query(request=_build_request(organization, use_case_id, query))
+
+    # TODO(layer): The layer should be able to make the totals query and the series query in one.
```

Comment:
> The layer should be able to make the totals query and the series query in one.

Can you elaborate on what you mean here? Why would the layer make two queries?

Reply:
Is this function fetching the timeseries and the totals for the same timeseries? That's something Clickhouse supports automatically (see the `WITH TOTALS` [clause](https://clickhouse.com/docs/en/sql-reference/statements/select/group-by#with-totals-modifier)). Is that what you're trying to do here?

Reply:
Exactly

Reply:
OK well then we can support adding the `WITH TOTALS` value to the query. That should return the totals in a single query.